src/CoreWCF.Primitives/tests/DependencyInjection/ServiceInstanceContextModeTests.cs

```diff
@@ -0,0 +1,174 @@
+﻿using CoreWCF.Channels;
+using CoreWCF.Description;
+using CoreWCF.Primitives.Tests.Helpers;
+using Helpers;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using Xunit;
+
+namespace DependencyInjection
+{
+    public class ServiceInstanceContextModeTests
+    {
+        [Fact]
+        public static void InstanceContextMode_Single()
+        {
+            SingleInstanceContextSimpleService.CreationCount = 0;
+            SingleInstanceContextSimpleService.DisposalCount = 0;
+            var services = new ServiceCollection();
+            var serviceInstance = new SingleInstanceContextSimpleService();
+            services.AddSingleton(serviceInstance);
+            TestHelper.BuildDispatcherAndCallService<SingleInstanceContextSimpleService>(services);
+            Assert.Equal(1, SingleInstanceContextSimpleService.CreationCount);
+            Assert.Equal(0, SingleInstanceContextSimpleService.DisposalCount);
+            Assert.Equal(1, serviceInstance.CallCount);
+        }
+
+        [Fact]
+        public static void InstanceContextMode_Single_NoInjection()
+        {
+            SingleInstanceContextSimpleService.CreationCount = 0;
+            SingleInstanceContextSimpleService.DisposalCount = 0;
+            var services = new ServiceCollection();
+            TestHelper.BuildDispatcherAndCallService<SingleInstanceContextSimpleService>(services);
+            Assert.Equal(1, SingleInstanceContextSimpleService.CreationCount);
+            Assert.Equal(0, SingleInstanceContextSimpleService.DisposalCount);
+        }
+
+        [Fact]
+        public static void InstanceContextMode_PerCall()
+        {
+            PerCallInstanceContextSimpleService.CreationCount = 0;
+            PerCallInstanceContextSimpleService.DisposalCount = 0;
+            var services = new ServiceCollection();
+            services.AddTransient<PerCallInstanceContextSimpleService>();
+            string serviceAddress = "http://localhost/dummy";
+            var serviceDispatcher = TestHelper.BuildDispatcher<PerCallInstanceContextSimpleService>(services, serviceAddress);
+            IChannel mockChannel = new MockReplyChannel(services.BuildServiceProvider());
+            var dispatcher = serviceDispatcher.CreateServiceChannelDispatcher(mockChannel);
+            // Instance created as part of service startup to probe if type is availale in DI
+            Assert.Equal(1, PerCallInstanceContextSimpleService.CreationCount);
+            Assert.Equal(1, PerCallInstanceContextSimpleService.DisposalCount);
+
+            PerCallInstanceContextSimpleService.CreationCount = 0;
```

Comment:
nit: test would be slightly easier to follow if you also reset DisposalCount

Reply:
not a nit, clear code is better code! The less you need to think about why something isn't how you would immediately expect, the better. I'll make the change.